components/Carousel/Carousel.js

```diff
@@ -1,8 +1,28 @@
 class Carousel {
+    constructor(carousel) {
+        this.carousel = carousel;
+        this.images = this.carousel.querySelectorAll("img")
+        this.rButton = this.carousel.querySelector(".right-button");
+        this.lButton = this.carousel.querySelector(".left-button");
+        this.currentIndex = 0;
+        currentImage = this.images[this.currentIndex].style.display = "block";
+        
+        this.rButton.addEventListener("click", () => this.toggleRight());
+        this.lButton.addEventListener("click", () => this.toggleLeft());
+    }
 
+    toggleRight() {
+
+    }
+
+    toggleLeft() {
```

Comment:
So close! Next time even if you go over the 3 hrs don't stop, finish what you started. We both had very similar Carousel classes so check out my toggle logic and see how you can apply it to your project:
```
    prev(){
        this.currentImg.style.display = 'none';
        if (this.counter === 0) {
            this.counter = this.imgTotal-1;
            this.currentImg = this.carouselItems[this.counter];
            this.currentImg.style.display = 'block';
        } else {
            this.counter -= 1;
            this.currentImg = this.carouselItems[this.counter];
            this.currentImg.style.display = 'block';
        }
    }

    next(){
        this.currentImg.style.display = 'none';
        if (this.counter === this.imgTotal-1) {
            this.counter = 0;
            this.currentImg = this.carouselItems[this.counter];
            this.currentImg.style.display = 'block';
        } else {
            this.counter += 1;
            this.currentImg = this.carouselItems[this.counter];            
            this.currentImg.style.display = 'block';
        }
    }
```